packages/frontend/src/redux/actions/account.js

```diff
@@ -544,6 +545,7 @@ export const refreshAccount = (basicData = false) => async (dispatch, getState)
     }
 
     dispatch(setLocalStorage(wallet.accountId));
+    dispatch(activeAccountActions.setAccountId({ accountId: wallet.accountId }));
```

Comment:
So if this now needs to be called before `setupAccountReducer`, can we just inject the reducers for the new account here instead of calling `addAccountReducer` in `wallet.saveAccount()`?

Reply:
Nope, `refreshAccount` is dispatched on every route change automatically and in many other situations. `addAccountReducer` needs to be called only in a situation when a new account is added.

I will work on introducing one action, that will be dispatched when the account creation/recovery is over, and then I will move `addAccountReducer` there. But it's not that easy as we have many methods for recovery or creating accounts, and it will be a part of separate PR.